static/js/xmleditor/xsd2json.js

```diff
@@ -306,36 +304,22 @@ SchemaManager.prototype.processIncludes = function(schema) {
 };
 
 SchemaManager.prototype.setOriginatingRoot = function() {
-
-	// Select root element if one is specified
-	if (this.options.rootElement != null) {
-		this.originatingRoot = this.originatingSchema.root;
-		
-		for (var index in this.originatingRoot.elements) {
-			var topLevelElement = this.originatingRoot.elements[index];
-			
-			if (this.options.rootElement != topLevelElement.name) {
-				this.originatingRoot = topLevelElement;
-			}
-		}
-	} else {
-		this.originatingRoot = {
-				schema : true,
-				ns : this.originatingSchema.targetNSIndex,
-				namespaces : [],
-				elements : [],
-				np : true
-			};
+	this.originatingRoot = {
+			schema : true,
+			ns : this.originatingSchema.targetNSIndex,
+			namespaces : [],
+			elements : [],
+			np : true
+		};
+	
```

Comment:
Yeah, that's some solid nesting, luckily the data its dealing with is generally pretty small (usually only 0 to 3 imports)